test/Microsoft.TemplateEngine.Authoring.TemplateVerifier.IntegrationTests/TemplateEngineSamplesTest.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Logging;
+using Microsoft.TemplateEngine.Authoring.TemplateApiVerifier;
+using Microsoft.TemplateEngine.TestHelper;
+using Microsoft.TemplateEngine.Tests;
+using Xunit.Abstractions;
+
+namespace Microsoft.TemplateEngine.Authoring.TemplateVerifier.IntegrationTests
+{
+    public class TemplateEngineSamplesTest : TestBase
+    {
+        private readonly ILogger _log;
+
+        public TemplateEngineSamplesTest(ITestOutputHelper log)
+        {
+            _log = new XunitLoggerProvider(log).CreateLogger("TestRun");
+        }
+
+        [Theory]
+        [InlineData("01-basic-template", "sample01", null, "no args")]
+        [InlineData("02-add-parameters", "sample02", new[] { "copyrightName", "Test Copyright", "title", "Test Title" }, "text args")]
+        [InlineData("03-optional-page", "sample03", new[] { "enableContactPage", "true" }, "optional content included")]
+        [InlineData("03-optional-page", "sample03", null, "optional content excluded")]
+        [InlineData("04-parameter-from-list", "sample04", new[] { "BackgroundColor", "dimgray" }, "the parameter from the list of options")]
+        [InlineData("05-multi-project", "sample05", new[] { "--includetest", "true" }, "the optional test project included")]
+        [InlineData("05-multi-project", "sample05", new[] { "--includetest", "false" }, "the optional test project excluded")]
```

Comment:
```suggestion
        [InlineData("05-multi-project", "sample05", new[] { "includetest", "true" }, "the optional test project included")]
        [InlineData("05-multi-project", "sample05", new[] { "includetest", "false" }, "the optional test project excluded")]
```

when testing through API, the parameter names should match the name in `template.json`, i.e. `--` is not needed.